pkg/builder/merged_entity_meta_data_builder.go

```diff
@@ -0,0 +1,344 @@
+	}
+	return nil
+}
+
+// =========================== MergedEntityMetadata_EntityField ===================================
+type fieldBuilder struct {
+	fieldName string
+	paths     []string
+}
+
+func newFieldBuilder(fieldName string) *fieldBuilder {
+	return &fieldBuilder{
+		fieldName: fieldName,
+	}
+}
+
+func (builder *fieldBuilder) build() *proto.MergedEntityMetadata_EntityField {
+	fieldName := builder.fieldName
+	if fieldName != "" {
+		entityFieldBuilder := &proto.MergedEntityMetadata_EntityField{}
+		entityFieldBuilder.FieldName = &fieldName
+		entityFieldBuilder.MessagePath = builder.paths
+		return entityFieldBuilder
+	}
+	return nil
+}
+
+// ============================== MergedEntityMetadata_CommodityBoughtMetadata =====================
+type commodityBoughtMetadata struct {
+	providerType     *proto.EntityDTO_EntityType
+	replacesProvider *proto.EntityDTO_EntityType
+	commBought       []proto.CommodityDTO_CommodityType
+}
+
+func newCommodityBoughtMetada(commBought []proto.CommodityDTO_CommodityType) *commodityBoughtMetadata {
+	return &commodityBoughtMetadata{
+		commBought: commBought,
+	}
+}
+
+// ============================== MergedEntityMetadataBuilder ======================================
+
+type MergedEntityMetadataBuilder struct {
+	// MergedEntityMetadata - the main protobuf structure to return
+	metadata *proto.MergedEntityMetadata
+
+	// MergedEntityMetadata consists of  MatchingMetadata for proxy and external entity
+	*matchingMetadataBuilder
+	propertyBuilders   []*propertyBuilder
+	fieldBuilders      []*fieldBuilder
+	commBoughtMetadata []*commodityBoughtMetadata
+}
+
+func NewMergedEntityMetadataBuilder() *MergedEntityMetadataBuilder {
+	builder := &MergedEntityMetadataBuilder{
+		metadata:                &proto.MergedEntityMetadata{},
+		matchingMetadataBuilder: newMatchingMetadataBuilder(),
+	}
+
+	return builder
+}
+
+func (builder *MergedEntityMetadataBuilder) build() (*proto.MergedEntityMetadata, error) {
+	metadata := &proto.MergedEntityMetadata{}
+
+	// Add the internal and external property matching metadata
+	matchingMetadata, err := builder.matchingMetadataBuilder.build()
+	fmt.Printf("******** %v\n", matchingMetadata)
+	if err != nil {
+		return nil, err
+	}
+	metadata.MatchingMetadata = matchingMetadata
+
+	if len(builder.commoditiesSold) > 0 {
+		metadata.CommoditiesSold = append(metadata.CommoditiesSold, builder.commoditiesSold...)
+	}
+
+	if len(builder.commBoughtMetadata) > 0 {
+		for _, commBought := range builder.commBoughtMetadata {
+			commMetadata := &proto.MergedEntityMetadata_CommodityBoughtMetadata{
+				CommodityMetadata: commBought.commBought,
+				ProviderType:      commBought.providerType,
+				ReplacesProvider:  commBought.replacesProvider,
+			}
+			metadata.CommoditiesBought = append(metadata.CommoditiesBought, commMetadata)
+		}
+	}
+	return metadata, nil
+}
+
+func (builder *MergedEntityMetadataBuilder) keepInTopology() *MergedEntityMetadataBuilder {
+	true_value := true
```

Comment:
This shouldn't be hardcoded right?  Should be configurable as before?  Also I think the default in Prometurbo is false for KeepStandalone.

Reply:
replaced with keepStandalone(bool bool_val)